From 45c37ffdd1eb1b97a6c1f0177ecf4a557f88428a Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Tue, 4 Jul 2017 16:31:56 +0800
Subject: [PATCH 2045/5242] MLK-15333-01 driver: clk: Fix clock round to a
 wrong rate issue

commit  1939cf3a712cc5b02672050c935b4a4c188b4e66 from
https://source.codeaurora.org/external/imx/linux-imx.git

The divider 'CLK_DIVIDER_ROUND_CLOSEST' flag should be enabled,
otherwise, the round clock will return a wrong clock rate.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk.h b/drivers/clk/imx/clk.h
index bc35053..0a2b901 100644
--- a/drivers/clk/imx/clk.h
+++ b/drivers/clk/imx/clk.h
@@ -195,7 +195,7 @@ static inline struct clk *imx_clk_divider2(const char *name, const char *parent,
 {
 	return clk_register_divider(NULL, name, parent,
 		CLK_SET_RATE_PARENT | CLK_SET_RATE_GATE | CLK_OPS_PARENT_ENABLE,
-			reg, shift, width, 0, &imx_ccm_lock);
+			reg, shift, width, CLK_DIVIDER_ROUND_CLOSEST, &imx_ccm_lock);
 }
 
 static inline struct clk *imx_clk_divider_flags(const char *name,
-- 
1.7.9.5

